installer/templates/phx_assets/package.json

```diff
@@ -13,12 +13,12 @@
     "@babel/core": "^7.0.0",
     "@babel/preset-env": "^7.0.0",
     "babel-loader": "^8.0.0",
-    "copy-webpack-plugin": "^4.5.0",
-    "css-loader": "^2.1.1",
-    "mini-css-extract-plugin": "^0.4.0",
+    "copy-webpack-plugin": "^5.1.1",
+    "css-loader": "^3.4.2",
+    "mini-css-extract-plugin": "^0.9.0",
     "optimize-css-assets-webpack-plugin": "^5.0.1",
-    "terser-webpack-plugin": "^1.1.0",
-    "webpack": "4.4.0",
+    "terser-webpack-plugin": "^2.3.2",
+    "webpack": "4.41.5",
     "webpack-cli": "^3.3.2"
```

Comment:
I think we would want a lock file update before merging as well!

Reply:
I think one of the good things about not having a lock file is that the packages are installed only based on the version constraints in `package.json`, and that would allow for the latest versions if this PR is merged. If we introduced a lock file, wouldn't that specify the exact versions to be installed? And in turn that we would have to update that lock-file far more often?

I'm not sure, I'm still a bit confused about how npm actually uses the lockfile (I usually use `npm ci` to get the exact lockfile dependencies, but I'm unsure how `npm install` handles a lockfile).

Reply:
After doing a bit of reading, I think that adding a `package-lock.json` file and running `npm install` would be the same as pinning all dependencies (and their dependencies) to specific versions. As far as I can see, the `package.json` file [will only be used as reference if it is newer](https://medium.com/coinmonks/everything-you-wanted-to-know-about-package-lock-json-b81911aa8ab8) than `package-lock.json`. There would be both advantages and disadvantages to this, I think.

Just for some stats: By running `npm ls | wc -l` inside a freshly installed Phoenix installation gives me 1377 different dependencies. (Removing deduped entried with `npm ls | grep -v deduped | wc -l` gives me 764). :open_mouth: 

An advantage to using a lockfile would be that we are very sure that a release of `phx_new` has good dependencies at the time of its release.

A disadvantage would be that this lockfile would very quickly specify old versions of packages, and also pretty quickly reference packages with known vulnerabilities. If a user runs `npm install` with an outdated lock-file, the user would get these exact dependencies. So we would have to make a new release of `phx_new` every time any dependency in the dependency tree needs to be updated. 

If there was no lockfile however, `npm install` would resolve the newest packages within the package version constraints, so it would fetch the newest versions possible. Then we would only have to make `phx_new` releases when the constraints no longer allow new packages (which is the current situation).

Reply:
I just found two further references:  [npm-install](https://docs.npmjs.com/cli-commands/install.html) [npm-package-lock.json](https://docs.npmjs.com/configuring-npm/package-lock-json)